### PR TITLE
Sort by status in symbols tab

### PR DESCRIPTION
**Pull request by mobile46**

### CODE REVIEW

File: src/gui/Src/Gui/SymbolView.cpp [hunk 1]
```diff
@@ -68,16 +68,23 @@ class ModuleStdTable final : public StdIconTable
...
         case MODSYMLOADING:
-            return tr("Loading");
+            text = tr("Loading");
+            break;
         case MODSYMLOADED:
-            return tr("Loaded");
+            text = tr("Loaded");
+            break;
         }
+        setCellContent(r, c, QString::number(status));
```
Comment (mrexodia) [inline src/gui/Src/Gui/SymbolView.cpp:86]: This should be set when the symbols are populated instead.

**Review by mrexodia — CHANGES_REQUESTED** [review]

**Comment by mrexodia** [comment]
What exactly doesn't work about sorting the status? It seems to work fine for me...

**Comment by mobile46** [comment]
No, it's not sorting correctly. It's sorting by the previously selected row. Please see attached gif.

x64dbg_HnYwYVGRkN

**Comment by mrexodia** [comment]
Lol my bad, I was testing the symbols tab 🤦‍♂️

**Comment by mrexodia** [comment]
I'll get back to this btw, need to verify something before merging it.

**Comment by mrexodia** [comment]
Finally had some time to investigate this and as I suspected there is an issue with the implementation. The sorting function always uses the text contents of the table, but in this case the table is 'virtual' and the symbol status isn't present. You add a number that represents the current status, but this is only done when the modules refresh.

If symbols are actually loaded, the displayed status is changed, but the number you set in the cell isn't updated. So this means you will just be sorting incorrectly in a different way.

Thanks for bringing this to my attention though, I will fix it and credit you in the commit.

**Pull request closed without merging**